tt_metal/impl/dispatch/memcpy.hpp

```diff
@@ -24,83 +45,98 @@ namespace tt::tt_metal {
 #if defined(__x86_64__) || defined(__i386__)
 template <bool debug_sync = false>
 static inline void memcpy_to_device(void* __restrict dst, const void* __restrict src, size_t n) {
```

Comment:
While we're cleaning this up, please consider
```suggestion
void memcpy_to_device(void* __restrict dst, const void* __restrict src, size_t n) {
```
`static` here means "internal linkage" which doesn't make sense for this function definition. `inline` is already implied because it's a `template`.